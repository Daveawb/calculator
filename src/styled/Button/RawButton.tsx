import * as React from 'react';

export type RawButtonProps = {
    children?: React.ReactNode;
    className?: string;
    fn: (event: React.MouseEvent) => void;
    value: string;
};

export const RawButton: React.FC<RawButtonProps> = (props: RawButtonProps) => {
    const { value, fn, className } = props;
    return (
        <button type="button" className={className} onClick={fn}>
            <span>{value}</span>
        </button>
    );
};
