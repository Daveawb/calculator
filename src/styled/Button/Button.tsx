import * as React from 'react';
import styled, { css } from 'styled-components';
import { RawButton, RawButtonProps } from './RawButton';

export type ButtonProps = RawButtonProps & {
    grow?: boolean;
    bg?: string;
    bgHover?: string;
};

export const Button: React.ComponentType<ButtonProps> = styled(RawButton)`
    display: flex;
    justify-content: center;
    min-width: 90px;
    min-height: 90px;
    margin: 4px;
    border-radius: 5px;
    color: #fff;
    cursor: pointer;
    font-size: 2rem;
    border: none;
    box-shadow: 1px 1px 3px rgba(1, 1, 1, 0.2);

    ${(props: ButtonProps) =>
        props.grow &&
        css`
            flex-grow: 1;
        `}

    ${(props: ButtonProps) => {
        const bg: string = props.bg || '#1795d4';
        const bgHover: string = props.bgHover || '#196b98';

        return css`
            background-color: ${bg};

            &:hover,
            &:focus {
                background-color: ${bgHover};
            }
        `;
    }}
`;
