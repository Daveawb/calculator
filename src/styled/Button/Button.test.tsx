import { shallow } from 'enzyme';
import * as React from 'react';
import { Button } from './Button';

describe('<Button /> styled component', () => {
    test('matches snapshot', () => {
        const fn = (event: React.MouseEvent) => undefined;
        const wrapper = shallow(<Button value="test" fn={fn} />);
        expect(wrapper).toMatchSnapshot();
    });
});
