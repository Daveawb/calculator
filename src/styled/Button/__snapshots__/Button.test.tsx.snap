// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> styled component matches snapshot 1`] = `
<StyledComponent
  fn={[Function]}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    display: flex;
    justify-content: center;
    min-width: 90px;
    min-height: 90px;
    margin: 4px;
    border-radius: 5px;
    color: #fff;
    cursor: pointer;
    font-size: 2rem;
    border: none;
    box-shadow: 1px 1px 3px rgba(1, 1, 1, 0.2);

    ",
          [Function],
          "

    ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(RawButton)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "self": [Circular],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
